### ISSUE TITLE
Understanding of network loss

### ISSUE THREAD

**Issue by adhara123007** [lmb-freiburg/flownet2#192]
I am working with the FlowNet2-CSS network. The network reports a loss as follows 

Sum_loss = net3_flow_loss2*0.005+net3_flow_loss3*0.01+net3_flow_loss4*0.02+net3_flow_loss5*0.08+ net3_flow_loss6*0.32

But when you are reporting the loss in the publication (FlowNet 2.0: Evolution of Optical Flow Estimation with Deep Networks), are you reporting this error or you are reporting only net3_flow_loss6?

@eddy-ilg @nikolausmayer 
Thank for being proactive on this forum.

**Comment by nikolausmayer** [comment]
Note that the printed losses in your output range from loss6 (lowest resolution) up to loss2 (quarter-resolution). Our reported errors are EPE/AAE evaluations, i.e. the network output is upsampled to full resolution (if it isn't already there) and compared to the groundtruth.

What I can't tell you right now is whether our FlowNet2-CSS errors are from the loss2 that you are seeing, or from a FlowNet2-CSS version which goes up to full-resolution "loss0". In the full "FlowNet2", the "FlowNet2-CSS" part only goes to quarter-resolution level (the final resolution isn't recovered until the "fuse" net which joins the -CSS and the -SD streams).

The FlowNet2-CSS training prototxt file in the repo makes most sense if you want to train it for inclusion into a full FlowNet2. If you want to use the FlowNet2-CSS on its own, it's probably a good idea to complete the upsampling steps back to full resolution.

**Comment by ADHARACSM** [comment]
I ran the FlowNet2-CSS with the deploy.prototxt (using the weights I get from training using the train.prototxt provided by you). I see I get full resolution. Does it mean the upsampling can be avoided?

**Comment by nikolausmayer** [comment]
@ADHARACSM The deploy prototxt file includes a final upsampling layer (type "Resample") which brings the flow output to the spatial resolution of the input, and then a "convolution" layer --- but that convolution is not learned, it only rescales the result to be a valid optical flow field (our architecture needs its input dimensions to be multiples of 64, so there is some rescaling done in deploy mode).

The output of this network will not be as good as a CSS network trained to full resolution (but it _is_ a valid full-resolution result).

**Comment by adhara123007** [comment]
Thanks again.
